build-system/tasks/helpers.js

```diff
@@ -665,7 +665,10 @@ async function compileJs(srcDir, srcFilename, destDir, options) {
     const watchFunc = async () => {
       await doCompileJs({...options, continueOnError: true});
     };
-    watch(deps).on('change', debounce(watchFunc, watchDebounceDelay));
+    watch(deps).on(
+      'change',
+      debounce(watchFunc, watchDebounceDelay, {immediate: true})
```

Comment:
There are 5 other watchers that use the same library:

https://github.com/ampproject/amphtml/blob/e6a551baf33ae69c3552c8d98c0037279edf9e78/build-system/tasks/3p-vendor-helpers.js#L45
https://github.com/ampproject/amphtml/blob/e6a551baf33ae69c3552c8d98c0037279edf9e78/build-system/tasks/analytics-vendor-configs.js#L51
https://github.com/ampproject/amphtml/blob/e6a551baf33ae69c3552c8d98c0037279edf9e78/build-system/tasks/extension-helpers.js#L435
https://github.com/ampproject/amphtml/blob/e6a551baf33ae69c3552c8d98c0037279edf9e78/build-system/tasks/serve.js#L236
https://github.com/ampproject/amphtml/blob/e6a551baf33ae69c3552c8d98c0037279edf9e78/build-system/tasks/css/index.js#L147

Can we update all of them in the same way? (Alternatively, you could add a helper somewhere in `build-system/common/`, but I'm not sure if it's worth the effort.)

Reply:
Certainly. I verified that all locations also made sense for leading/trailing edge invocations as well.
I initially placed the helper in `common/utils`, but that led to circular dependency warnings, so I created a new file to avoid them.

```
(node:18012) Warning: Accessing non-existent property 'debounce' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18012) Warning: Accessing non-existent property 'debounce' of module exports inside circular dependency
(node:18012) Warning: Accessing non-existent property 'debounce' of module exports inside circular dependency
(node:18012) Warning: Accessing non-existent property 'debounce' of module exports inside circular dependency
```